lib/lambda_ethereum_consensus/beacon/pending_blocks.ex

```diff
@@ -199,12 +186,18 @@ defmodule LambdaEthereumConsensus.Beacon.PendingBlocks do
 
         # If all the other conditions are false, add block to fork choice
         true ->
-          ForkChoice.on_block(block_info)
-          state |> Map.put(block_root, {signed_block, :processing})
+          process_block(state, block_info)
       end
     end)
   end
 
+  defp process_block(state, block_info) do
+    case ForkChoice.on_block(block_info) do
```

Comment:
We should run process blocks afterwards, as there may be blocks that now can be processed.